clientlibrary/worker/worker.go

```diff
@@ -75,6 +78,9 @@ func NewWorker(factory kcl.IRecordProcessorFactory, kclConfig *config.KinesisCli
 		mService = metrics.NoopMonitoringService{}
 	}
 
+	// Create a pseudo-random number generator and seed it.
+	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
This may not be needed. Since we are using:
// Intn returns, as an int, a non-negative pseudo-random number in [0,n)
// from the default Source.
// It panics if n <= 0.
func Intn(n int) int { return globalRand.Intn(n) }

Reply:
By not seeding the random number generator, the default seed of 0 is used. This would led to the same sequence of random numbers to be generated on all workers, annihilating or at least greatly reducing the effect of the jitter.

Reply:
I didn's see rng is used at all. The later call uses:
rand.Intn()

Reply:
You're totally right. I just updated the PR, now using the seeded random number generator.